### PR TITLE
Associate "firestore.rules" filename with Firebase Firestore icon

### PR DESCRIPTION
**Pull request by donmckenna**
**Changes proposed:**

- [x] Add

### CODE REVIEW

**Comment by JimiC** [comment]
Can you please post a link so we can get more info on the filename? I suspect that this may need its own icon.

**Comment by donmckenna** [comment]
Cloud Firestore (Firestore) is a NoSQL database under the Firebase family of products.
https://firebase.google.com/products/firestore/
`firestore.rules` are the security rules for Firestore database access.

I agree with you and have made a Firestore-specific icon based on the existing `firebase.svg` icon.
It retains the Firebase root but includes the child Firestore logo in the corner like the TS or JS on Angular icons.

Hope this is more useful.

File: src/icon-manifest/supportedExtensions.ts [hunk 1]
```diff
@@ -226,6 +226,7 @@ export const extensions: IFileCollection = {
     { icon: 'favicon', extensions: ['favicon.ico'], filename: true, format: FileFormat.svg },
     { icon: 'fbx', extensions: ['fbx'], format: FileFormat.svg },
     { icon: 'firebase', extensions: ['.firebaserc'], filename: true, format: FileFormat.svg },
+    { icon: 'firebase_firestore', extensions: ['firestore.rules'], filename: true, format: FileFormat.svg },
```
Comment (JimiC) [inline src/icon-manifest/supportedExtensions.ts]: Better use simply `firestore` as the `icon` attribute value.

Comment (donmckenna) [reply src/icon-manifest/supportedExtensions.ts]: Changed :)

Comment (donmckenna) [reply src/icon-manifest/supportedExtensions.ts]: Not that it needs to change again, but I'll just say there is no Firestore on its own without Firebase, which is why I included `firebase_` in the initial `icon` attribute value.
I also prefer the more succinct `firestore`, but just wanted to make the distinction that Firestore is a sort of child of Firebase, if that affects any existing patterns `vscode-icons` may be already using.

Comment (donmckenna) [reply src/icon-manifest/supportedExtensions.ts]: Fair enough.

**Comment by donmckenna** [comment]
Thank you for doing that. I will be more observant next time.

**Comment by donmckenna** [comment]
Thank you! And thank you for a great plugin :)

**Pull request merged**